lib/bandit/http1/handler.ex

```diff
@@ -25,7 +25,16 @@ defmodule Bandit.HTTP1.Handler do
       if Keyword.get(state.opts.http_1, :clear_process_dict, true), do: clear_process_dict()
       gc_every_n_requests = Keyword.get(state.opts.http_1, :gc_every_n_keepalive_requests, 5)
       if rem(requests_processed, gc_every_n_requests) == 0, do: :erlang.garbage_collect()
-      {:continue, Map.put(state, :requests_processed, requests_processed)}
+
+      state = Map.put(state, :requests_processed, requests_processed)
+
+      # We have bytes that we've read but haven't yet processed, tail call handle_data to start
+      # reading the next request
+      if IO.iodata_length(transport.buffer) != 0 do
```

Comment:
```suggestion
      if transport.buffer != <<>> do
```

One small suggestion with the following compare `bytes != <<>>` and `IO.iodata_length(bytes) != 0`:

```
Name                                             ips        average  deviation         median         99th %
bytes != <<>> with large bytes              216.30 M        4.62 ns  ±6611.15%           0 ns          11 ns
bytes != <<>> with empty bytes              199.46 M        5.01 ns  ±6209.18%           0 ns          11 ns
bytes != <<>> with small bytes              189.55 M        5.28 ns  ±6057.20%           0 ns          13 ns
IO.iodata_length != 0 with large bytes       39.31 M       25.44 ns ±34060.36%          15 ns          41 ns
IO.iodata_length != 0 with samll bytes       37.95 M       26.35 ns ±39355.33%          16 ns          44 ns
IO.iodata_length != 0 with empty bytes       35.85 M       27.89 ns ±33175.52%          16 ns          45 ns

Comparison:
bytes != <<>> with large bytes              216.30 M
bytes != <<>> with empty bytes              199.46 M - 1.08x slower +0.39 ns
bytes != <<>> with small bytes              189.55 M - 1.14x slower +0.65 ns
IO.iodata_length != 0 with large bytes       39.31 M - 5.50x slower +20.82 ns
IO.iodata_length != 0 with samll bytes       37.95 M - 5.70x slower +21.73 ns
IO.iodata_length != 0 with empty bytes       35.85 M - 6.03x slower +23.27 ns
```

Reply:
I'm deliberately not doing this because 1. It's almost always the case that this is empty (it's not not if the client is pipelining, which is pretty rare), and 2. since it's an iolist there's uncountably many ways that it could be empty (`<<>>`, `[]`, [<<>>], `[<<>>, []]`, etc). So there's no performance hit in the most common case of it being empty, and it's always correct.

Reply:
Thanks very much for clarification.